template/docs/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,17 @@
+Provide a narrative description of what you are trying to accomplish:
+
+- Are you fixing a bug?
+  - Detail how the bug is invoked currently.
+  - Detail the original, incorrect behavior.
+  - Detail the new, expected behavior.
+
+- Are you creating a new feature?
+  - Why is the new feature needed? What purpose does it serve?
+  - How will users use the new feature?
+
```

Comment:
- Make sure you add only 1 new feature, multiple new features should be split in multiiple PR's.
  - Add tests for the new feature.